crates/proto/src/op/message.rs

```diff
@@ -589,7 +590,7 @@ impl Message {
                 query_count: self.queries.len(),
                 answer_count: self.answers.len(),
                 nameserver_count: self.name_servers.len(),
-                additional_count: self.additionals.len(),
+                additional_count: self.additionals.len() + if self.edns.is_some() { 1 } else { 0 },
```

Comment:
EDNS is converted to an additional record, but on the Rust side not represented as one. So the counts here are off if this is not accounted for.